subsys/net/ip/net_context.c

```diff
@@ -2401,9 +2401,21 @@ static int recv_udp(struct net_context *context,
 
 	ARG_UNUSED(timeout);
 
+	/* If the context already has a connection handler, it means it's
+	 * already registered. In that case, all we have to do is 1) update
+	 * the callback registered in the net_context and 2) update the
+	 * user_data using net_conn_change_callback().
+	 *
+	 * Since net_conn_change_callback() takes the _other_ callback
+	 * function, net_context_packet_received() (I know it's confusing),
+	 * this must be the same function passed in net_conn_register() down below.
```

Comment:
This comment needs some clarification, it is a bit difficult to follow, could you formulate it better. Also remove the `(I know it's confusing)` sentence.

Reply:
Okay. I will update the comment.